pkg/textparse/promparse.go

```diff
@@ -370,12 +371,20 @@ func (p *PromParser) Next() (Entry, error) {
 }
 
 func (p *PromParser) parseLVals() error {
+	labels := [][]byte{}
 	t := p.nextToken()
 	for {
 		switch t {
 		case tBraceClose:
 			return nil
 		case tLName:
+			label := p.l.b[p.l.start:p.l.i]
+			for _, l := range labels {
```

Comment:
This is n^2, probably better to do it in the scrape code.

Reply:
if we do in the scrape code, our parser will not be open metrics complicant and will not be reusable by others. WDYT?

Reply:
I could extract that in a common function between the two parsers.

Reply:
It is also hard/not that mych added value to share that code between parsers as it is using private apis.

Reply:
This is our internal parser for speed, not an official OM parser. It's not meant to be reused by others, as it's tightly coupled to the scrape code.

Reply:
I understand, and I did not know it. I was hoping that our om parser could be reused by 3rd party implementations. Also getting the code here will help when we will implement the openmetrics reference test suite.

Reply:
The reference test suite is in Python.

This is never intended to be a fully correct OM parser that detects all errors, just a good enough parser that will hopefully catch the main errors.